### PR TITLE
Support AudioMNIST

### PR DESCRIPTION
**Pull request by csukuangfj**
# Usage
```bash
git clone https://github.com/soerenab/AudioMNIST
cd AudioMNIST
lhotse prepare audio-mnist . ./abc
cd abc

lhotse cut simple -r ./audio_mnist_recordings.jsonl.gz -s ./audio_mnist_supervisions.jsonl.gz audio_mnist_cuts.jsonl.gz

ls -lh

total 5912
-rw-r--r--  1 fangjun  staff   1.0M Jun 17 16:26 audio_mnist_cuts.jsonl.gz
-rw-r--r--  1 fangjun  staff   419K Jun 17 16:24 audio_mnist_recordings.jsonl.gz
-rw-r--r--  1 fangjun  staff   376K Jun 17 16:24 audio_mnist_supervisions.jsonl.gz

lhotse cut describe ./audio_mnist_cuts.jsonl.gz
```

```
Cut statistics:
╒═══════════════════════════╤══════════╕
│ Cuts count:               │ 30000    │
├───────────────────────────┼──────────┤
│ Total duration (hh:mm:ss) │ 05:21:18 │
├───────────────────────────┼──────────┤
│ mean                      │ 0.6      │
├───────────────────────────┼──────────┤
│ std                       │ 0.1      │
├───────────────────────────┼──────────┤
│ min                       │ 0.3      │
├───────────────────────────┼──────────┤
│ 25%                       │ 0.6      │
├───────────────────────────┼──────────┤
│ 50%                       │ 0.6      │
├───────────────────────────┼──────────┤
│ 75%                       │ 0.7      │
├───────────────────────────┼──────────┤
│ 99%                       │ 0.9      │
├───────────────────────────┼──────────┤
│ 99.5%                     │ 1.0      │
├───────────────────────────┼──────────┤
│ 99.9%                     │ 1.0      │
├───────────────────────────┼──────────┤
│ max                       │ 1.0      │
├───────────────────────────┼──────────┤
│ Recordings available:     │ 30000    │
├───────────────────────────┼──────────┤
│ Features available:       │ 0        │
├───────────────────────────┼──────────┤
│ Supervisions available:   │ 30000    │
╘═══════════════════════════╧══════════╛
SUPERVISION custom fields:
Speech duration statistics:
╒══════════════════════════════╤══════════╤══════════════════════╕
│ Total speech duration        │ 05:21:18 │ 100.00% of recording │
├──────────────────────────────┼──────────┼──────────────────────┤
│ Total speaking time duration │ 05:21:18 │ 100.00% of recording │
├──────────────────────────────┼──────────┼──────────────────────┤
│ Total silence duration       │ 00:00:01 │ 0.00% of recording   │
╘══════════════════════════════╧══════════╧══════════════════════╛
```

### CODE REVIEW

File: lhotse/recipes/audio_mnist.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+collected in quiet offices with a RØDE NT-USB microphone as
+mono channel signal at a sampling frequency of 48kHz and were
+saved in 16 bit integer format. In addition to audio recordings, meta
+information including age (range: 22 - 61 years), gender (12 female
+/ 48 male), origin and accent of all speakers were collected as well.
+All speakers were informed about the intent of the data collection
+and have given written declarations of consent for their participation prior
+to their recording session.
+
+We don't provide download_audio_mnist().
```
Comment (pzelasko) [inline lhotse/recipes/audio_mnist.py:16]: technically we could provide it and let it call `subprocess.run("git clone ...", shell=True)`, WDYT?

**Review by pzelasko — APPROVED** [review]
LGTM, I just left one comment -- up to you

File: lhotse/recipes/audio_mnist.py (hunk 1 above)
Comment (rouseabout) [reply lhotse/recipes/audio_mnist.py:16]: github produces tarballs automatically: https://github.com/soerenab/AudioMNIST/archive/master.tar.gz

Comment (csukuangfj) [reply lhotse/recipes/audio_mnist.py:16]: > github produces tarballs automatically: https://github.com/soerenab/AudioMNIST/archive/master.tar.gz

Thanks! Will add the download command.

Comment (pzelasko) [reply lhotse/recipes/audio_mnist.py:16]: Should I just merge it? We can always add download later if needed.

**Comment by csukuangfj** [comment]
# Usage of `lhotse download audio-mnist`

```
(py38) fangjuns-MacBook-Pro:lhotse fangjun$ lhotse download audio-mnist
Usage: lhotse download audio-mnist [OPTIONS] TARGET_DIR
Try 'lhotse download audio-mnist --help' for help.

Error: Missing argument 'TARGET_DIR'.


(py38) fangjuns-MacBook-Pro:lhotse fangjun$ lhotse download audio-mnist --help
Usage: lhotse download audio-mnist [OPTIONS] TARGET_DIR

  AudioMNIST dataset download.

Options:
  --force-download BOOLEAN  If True, download even if file is present.
                            [default: False]
  --help                    Show this message and exit.  [default: False]

(py38) fangjuns-MacBook-Pro:lhotse fangjun$ lhotse download audio-mnist ./abc
abc/master.tar.gz: 991MB [13:12, 1.25MB/s]
2023-07-30 23:12:58,131 INFO [audio_mnist.py:64] Untarring master.tar.gz.

(py38) fangjuns-MacBook-Pro:lhotse fangjun$ ls -lh abc
total 1966208
drwxrwxr-x  10 fangjun  staff   320B Oct 24  2018 AudioMNIST
-rw-r--r--   1 fangjun  staff   945M Jul 30 23:12 master.tar.gz

(py38) fangjuns-MacBook-Pro:lhotse fangjun$ ls -lh abc/AudioMNIST/
total 48
-rw-rw-r--   1 fangjun  staff   1.0K Oct 24  2018 LICENSE
-rw-rw-r--   1 fangjun  staff   1.9K Oct 24  2018 README.md
drwxrwxr-x  63 fangjun  staff   2.0K Oct 24  2018 data
drwxrwxr-x   4 fangjun  staff   128B Oct 24  2018 models
-rw-rw-r--   1 fangjun  staff   9.3K Oct 24  2018 preprocess_data.py
drwxrwxr-x   6 fangjun  staff   192B Oct 24  2018 recording_scripts
-rw-rw-r--   1 fangjun  staff   876B Oct 24  2018 requirements.txt
```

File: lhotse/recipes/audio_mnist.py (hunk 1 above)
Comment (csukuangfj) [reply lhotse/recipes/audio_mnist.py]: Sorry for the late reply. I just added it.

**Review by pzelasko — APPROVED** [review]

**Pull request merged**